GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCOREvent+GDTCCTSupport.m

```diff
@@ -0,0 +1,194 @@
+      if (error) {
+        self.customBytes = nil;
+        GDTCORLogDebug(@"Error when setting an event's network_connection_info: %@", error);
+      }
+    }
+  } @catch (NSException *exception) {
+    GDTCORLogDebug(@"Error when setting an event's network_connection_info: %@", exception);
+  }
+}
+
+- (nullable NSData *)networkConnectionInfoData {
+  if (self.customBytes) {
+    @try {
+      NSError *error;
+      NSDictionary *bytesDict = [NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                                options:0
+                                                                  error:&error];
+      NSString *base64Data = bytesDict[GDTCCTNetworkConnectionInfo];
+      NSData *networkConnectionInfoData = [[NSData alloc] initWithBase64EncodedString:base64Data
+                                                                              options:0];
+      if (error) {
+        GDTCORLogDebug(@"Error when getting an event's network_connection_info: %@", error);
+        return nil;
+      } else {
+        return networkConnectionInfoData;
+      }
+    } @catch (NSException *exception) {
+      GDTCORLogDebug(@"Error when getting an event's network_connection_info: %@", exception);
+    }
+  }
+  return nil;
+}
+
+
+- (NSNumber *)eventCode {
+  if (self.customBytes) {
+    @try {
+      NSError *error;
+      NSDictionary *bytesDict = [NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                                options:0
+                                                                  error:&error];
+      NSString *eventCodeString = bytesDict[GDTCCTEventCodeInfo];
+
+      if (!eventCodeString) {
+        return nil;
+      }
+
+      NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
+      formatter.numberStyle = NSNumberFormatterDecimalStyle;
+      NSNumber *eventCode = [formatter numberFromString:eventCodeString];
+
+      if (error) {
+        GDTCORLogDebug(@"Error when getting an event's network_connection_info: %@", error);
+        return nil;
+      } else {
+        return eventCode;
+      }
+    } @catch (NSException *exception) {
+      GDTCORLogDebug(@"Error when getting an event's event_code: %@", exception);
+    }
+  }
+  return nil;
+}
+
+- (void) setEventCode:(NSNumber *)eventCode {
+  @try {
+    NSMutableDictionary *bytesDict;
+    NSError *error;
+    if (self.customBytes) {
+      bytesDict = [[NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                   options:0
+                                                     error:&error] mutableCopy];
+      if (error) {
+        GDTCORLogDebug(@"Error when setting an even'ts event_code: %@", error);
+        return;
+      }
+    } else {
+      bytesDict = [[NSMutableDictionary alloc] init];
+    }
+
+    [bytesDict setObject:[eventCode stringValue] forKey:GDTCCTEventCodeInfo];
```

Comment:
Probably check [eventCode stringValue] for nil, because setting a value to nil will cause a crash